frontend/src/pages/CompareV2.tsx

```diff
@@ -303,15 +389,44 @@ function MetricsDropdown(props: MetricsDropdownProps) {
   }
 
   return (
-    <>
-      <TwoLevelDropdown
-        title={`Choose a first ${metricsTabText} artifact`}
-        items={dropdownItems}
-        selectedItem={selectedItem}
-        setSelectedItem={setSelectedItem}
-      />
-      <VisualizationPlaceholder metricsTabText={metricsTabText} />
-    </>
+    <table>
+      <tbody>
+        <tr>
+          <td className={classes(css.cell, css.leftCell)}>
+            <TwoLevelDropdown
+              title={`Choose a first ${metricsTabText} artifact`}
+              items={dropdownItems}
+              selectedItem={firstSelectedItem}
+              setSelectedItem={setFirstSelectedItem}
```

Comment:
The `useEffect` and `useCallback` logics you use can actually be embedded into this callback:

```
updateSelectedItem={(selectedItem) => {
   // call setSelectedItem
   // call setSelectedArtifacts
}}

Reply:
That sounds good - done! https://github.com/kubeflow/pipelines/pull/7966/commits/17319488f91a28a31916e2c2425f5201bbdab853